core/src/main/java/org/apache/iceberg/formats/WriteBuilder.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.formats;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import org.apache.iceberg.FileContent;
+import org.apache.iceberg.MetricsConfig;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.io.FileAppender;
+
+/**
+ * Builder interface for creating file writers across supported data file formats. The {@link
+ * FormatModel} implementations provide the appropriate {@link WriteBuilder} instances.
+ *
+ * <p>The {@link WriteBuilder} follows the builder pattern to configure and create {@link
+ * FileAppender} instances that write data to the target output files.
+ *
+ * <p>This interface is directly exposed to users for parameterizing when only an appender is
+ * required.
+ */
+public interface WriteBuilder {
+  /** Set the file schema. */
+  WriteBuilder schema(Schema schema);
+
+  /**
+   * Sets the input schema accepted by the writer. If not provided derived from the {@link
+   * #schema(Schema)}.
+   */
+  WriteBuilder inputSchema(Object schema);
+
+  /**
+   * Set a writer configuration property which affects the writer behavior.
```

Comment:
nit: ReadBuilder comment mentions that unknown config is ignored. Should we mention it here (and for setAll) too?